app/models/amazon_agent_manager.rb

```diff
@@ -0,0 +1,393 @@
+require 'aws-sdk'
+require 'amazon_ssa_support'
+require 'yaml'
+require 'open3'
+
+class AmazonAgentManager
```

Comment:
I don't like `Manager` in the name because `Manager` is what we call EMSes elsewhere and this is going to get confusing.  Perhaps `AmazonAgentOrchestrator`?

Reply:
Another naming option is `AmazonAgentCoordinator`

Reply:
Can this code live in the amazon plugin instead of core?